nixos/modules/services/backup/bupstash.nix

```diff
@@ -0,0 +1,556 @@
+in
+{
+  meta.maintainers = [ lib.maintainers.thubrecht ];
+
+  options.services.bupstash = {
+    package = mkPackageOption pkgs "bupstash" { };
+
+    repositories = {
+      enable = mkEnableOption "bupstash repositories that can be accessed through ssh.";
+
+      user = mkOption {
+        type = str;
+        default = defaultRepositoryUser;
+        description = "User used for connection to the bupstash repository.";
+      };
+
+      group = mkOption {
+        type = str;
+        default = defaultRepositoryUser;
+        description = "Group of the user used for connection to the bupstash repository.";
+      };
+
+      access = mkOption {
+        type = listOf (
+          submodule {
+            options = {
+              repo = mkOption {
+                type = str;
+                description = "Name of the repository.";
+              };
+
+              keys = mkOption {
+                type = listOf str;
+                default = [ ];
+                description = "List of ssh keys that can be used to access this repository.";
+              };
+
+              allowed = mkOption {
+                type = listOf (
+                  enum [
+                    "init"
+                    "put"
+                    "list"
+                    "get"
+                    "remove"
+                    "gc"
+                    "sync"
+                  ]
+                );
+                default = [ ];
+                description = ''
+                  List of allowed bupstash actions through the ssh connection.
+                  If none are specified then all actions are authorized.
+                '';
+              };
+            };
+          }
+        );
+        default = { };
+        description = "List of ssh keys with access to the repository.";
+      };
+
+      home = mkOption {
+        type = passwdEntry path;
+        default = "/var/lib/${defaultRepositoryUser}";
+        description = "The path where the repositories will be stored.";
+      };
+
+      restrictCommands = mkOption {
+        type = bool;
+        default = true;
+        description = "If set to true, restricts the commands that can be run through ssh by the user to `bupstash ...`.";
+      };
+    };
+
+    jobs = mkOption {
+      type = attrsOf (
+        submodule (
+          { name, ... }:
+          {
+            options = {
+              paths = mkOption {
+                type = listOf path;
+                default = [ ];
+                description = "List of paths to backup.";
+              };
+
+              excludes = mkOption {
+                type = listOf str;
+                default = [ ];
+                description = "List of patterns to exlude from the backup.";
+              };
+
+              command = mkOption {
+                type = nullOr str;
+                default = null;
+                description = "Command used for getting the backup data.";
+              };
```

Comment:
Add an example command, e.g. `${lib.getBin config.services.postgresql.package}/bin/pgdump mydb` and mention that this is used to back up the stdout of the command.